### PR TITLE
Gen2 common objdet

### PR DESCRIPTION
**Pull request by jonngai**

### CODE REVIEW

File: include/depthai-shared/datatype/DetectionNetworkTypeEnum.hpp [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#pragma once
+
+#include <cstdint>
+
+enum DetectionNetworkTypeEnum { dntype_yolo, dntype_mobilenet };
```
Comment (themarpe) [inline include/depthai-shared/datatype/DetectionNetworkTypeEnum.hpp:5]: Move and rename the file to: `include/depthai-shared/pb/common/DetectionNetworkType.hpp`

And modify to enum class:
```
enum class DetectionNetworkType { YOLO, MOBILENET };
```

File: include/depthai-shared/pb/properties/CommonObjDetProperties.hpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#pragma once
```
Comment (themarpe) [inline include/depthai-shared/pb/properties/CommonObjDetProperties.hpp:1]: CommonObjDetProperties.hpp file still needed?

**Review by themarpe — CHANGES_REQUESTED** [review]

File: include/depthai-shared/pb/properties/CommonObjDetProperties.hpp (hunk 2 above)
Comment (jonngai) [reply include/depthai-shared/pb/properties/CommonObjDetProperties.hpp]: thanks. deleted.

File: include/depthai-shared/datatype/DetectionNetworkTypeEnum.hpp (hunk 1 above)
Comment (jonngai) [reply include/depthai-shared/datatype/DetectionNetworkTypeEnum.hpp]: sure. done.

**Review by themarpe — APPROVED** [review]

**Review by themarpe — APPROVED** [review]
Feel free to merge after other PRs are resolved as well

File: include/depthai-shared/pb/properties/DetectionNetworkProperties.hpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+#include "NeuralNetworkProperties.hpp"
+
+namespace dai {
+
+/**
+ * Properties for DetectionNetwork
+ */
+struct DetectionNetworkProperties : NeuralNetworkProperties {
+    std::string streamName;
```
Comment (themarpe) [inline include/depthai-shared/pb/properties/DetectionNetworkProperties.hpp:15]: I think this is redundant?

**Review by themarpe — CHANGES_REQUESTED** [review]
In DatatypeEnum.cpp change `hierarchy` to:
```
const std::unordered_map<DatatypeEnum, std::vector<DatatypeEnum>> hierarchy = {
    {DatatypeEnum::Buffer, {DatatypeEnum::ImgFrame, DatatypeEnum::NNData, DatatypeEnum::ImageManipConfig, DatatypeEnum::ImgDetections, DatatypeEnum::CameraControl}},
    {DatatypeEnum::ImgFrame, {}},
    {DatatypeEnum::NNData, {}},
    {DatatypeEnum::ImageManipConfig, {}},
    {DatatypeEnum::ImgDetections, {}},
    {DatatypeEnum::CameraControl, {}}};
```

**Pull request merged**